docs/extensibility/roslyn-version-support.md

```diff
@@ -20,7 +20,16 @@ As an example, to ensure that your custom analyzer works on all versions of Visu
 
 | Roslyn package version | Minimum supported Visual Studio version |
```

Comment:
@jmarolf Can we have a third column for Minimum supported SDK version in the case of command-line builds?

Reply:
.NET SDK versions do not line up in any real way with VS versions. We would need many more columns to explain this.

in SDK version 2.1.500, `2.1` referes to the maximum runtime version that this sdk can build against and `500` means the fifth release of the SDK.

Reply:
Isn't the minimum SDK version the same one ships with the specific Visual Studio version?
For example for Roslyn packages 3.9.x it's Visual Studio 16.9 and .NET SDK 5.0.200?

Reply:
hmm, so if you got to dot.net and download the .NET 5 SDK it will be 5.0.103 which is compatible with 3.8.3 compilers and newer. If you use the SDK that is installed by VS 2019 16.9 you get 5.0.200 which is compatible with 3.9.0. So If you just want to know what SDK ships with which Visual Studio that's an easy mapping. If you want to know which compiler to target so users of a particular SDK (say .NET 5 SDKs)  that becomes a more complex question that frankly needs a different table altogether.

Reply:
> If you want to know which compiler to target so users of a particular SDK (say .NET 5 SDKs) that becomes a more complex question that frankly needs a different table altogether.

Yeah this is what I meant. But I thought it's the same as "see the roslyn version supported by whatever VS version that shipped this particular SDK".

Now I'm more confused to be honest.

To be more clear, the intent here is when authoring analyzers. When I upgrade roslyn version in my analyzers:

- What VS version will be required for analyzer consumers.
- What SDK version will be required for analyzer consumers in command-line builds.

Reply:
I get the problem we need to solve here (and we do need something). We may want a different doc page for analyzer authors as things get complicated.